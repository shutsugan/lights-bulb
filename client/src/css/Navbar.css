.Navbar {
	display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    height: 50px;
    padding: 0 0 0 16px;
    background: var(--primary-color);
    font-size: .9em;
    color: #FFF;
}