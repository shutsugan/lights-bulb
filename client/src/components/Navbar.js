import React from 'react';

import '../css/Navbar.css'

const Navbar = props => (
	<div className="Navbar">
		<div>{props.date}</div>
		<div>{props.time}</div>

		{props.children}
	</div>
);

export default Navbar;