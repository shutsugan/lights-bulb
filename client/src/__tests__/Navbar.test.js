import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Navbar from '../components/Navbar';

import dateFormat from 'dateformat';

const now = new Date();
const date = dateFormat(now, 'dddd, dS mmmm, yyyy');
const time = dateFormat(now, 'h:MM TT');

describe('Navbar', () => {
    it('should be defined', () => {
      expect(Navbar).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = shallow(<Navbar date={date} time={time} />);
        expect(tree).toMatchSnapshot();
    });

    it('should return the same data and time', () => {
        const tree = shallow(<Navbar date={date} time={time} />);

        expect(tree.unrendered.props.date).toEqual(date);
        expect(tree.unrendered.props.time).toEqual(time);
      });
});