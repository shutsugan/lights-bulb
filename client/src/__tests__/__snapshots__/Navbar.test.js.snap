// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="Navbar"
>
    <div>
        Monday, 23rd July, 2018
    </div>
    <div>
        11:04 PM
    </div>
</div>,
  "nodes": Array [
    <div
      className="Navbar"
>
      <div>
            Monday, 23rd July, 2018
      </div>
      <div>
            11:04 PM
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Navbar
      date="Monday, 23rd July, 2018"
      time="11:04 PM"
/>,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="Navbar"
>
      <div>
            Monday, 23rd July, 2018
      </div>
      <div>
            11:04 PM
      </div>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Navbar
    date="Monday, 23rd July, 2018"
    time="11:04 PM"
/>,
}
`;
